pkg/planner/core/operator/logicalop/expression_util.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
```

Comment:
how about putting it into constraint pkg

Reply:
it cannot put it constraint pkg.it will happen import cycle.

```
        imports github.com/pingcap/tidb/pkg/planner/core/operator/logicalop
        imports github.com/pingcap/tidb/pkg/planner/core/constraint
        imports github.com/pingcap/tidb/pkg/planner/core/operator/logicalop: import cycle not allowed
```

Reply:
I also try to move this into ```planner/core```. but it still raise the ```import cycle```.